Lower calibration thread priority

arntanguy opened the pull request:
The calibration thread inherits the real-time thread's priority. As this is a computationally intensive thread, this causes the controller to freeze while the calibration is being computed, which can take several seconds (or more if it does not converge).

This PR reduces the thread priority to avoid such issues.
This was tested on `hrp2012c`, I will merge when pipeline succeeds.

arntanguy committed:
Lower calibration thread priority

arntanguy committed:
[clang] Fix formatting

arntanguy committed:
[ci] Upgrade clang-format

gergondet commented on src/states/RunCalibrationScript.cpp:
```diff
@@ -73,6 +73,15 @@ void RunCalibrationScript::start(mc_control::fsm::Controller & ctl_)
     }
     completed_ = true;
   });
+  // Lower thread priority so that it has a lesser priority than the real time
+  // thread
+  auto th_handle = th_.native_handle();
+  int policy = 0;
+  sched_param param{};
+  pthread_getschedparam(th_handle, &policy, &param);
+  mc_rtc::log::info("[{}] Calibration thread priority: {}", name(), param.sched_priority);
+  param.sched_priority = 80;
+  pthread_setschedparam(th_handle, SCHED_RR, &param);
```
This should be surrounded with `#ifndef WIN32` (or something better that checks that `std::thread` implementation is pthread based)

gergondet replied:
This call might fail if the user doesn't have permissions to set such high priority (but in that case it's likely not running in a real-time system). In any case, we shouldn't ignore the call result and display a message if appropriate

gergondet reviewed (changes requested):
LGTM except for the Windows compatibility and the error message for failing to set a lower priority

arntanguy committed:
[thread] Do not lower priority on windows systems

arntanguy commented:
I added the check for windows system (I'm not sure how one would ensure that a thread implementation is pthread-based), and added an error message in case it fails.
